src/main/java/net/snowflake/client/core/arrow/BigIntToTimeConverter.java

```diff
@@ -36,29 +36,34 @@ public BigIntToTimeConverter(
    */
   private SFTime toSFTime(int index) {
     long val = bigIntVector.getDataBuffer().getLong(index * BigIntVector.TYPE_WIDTH);
-    return SFTime.fromFractionalSeconds(val, context.getScale(columnIndex));
+    return toSFTime(val, context.getScale(columnIndex));
+  }
+
+  private static SFTime toSFTime(long val, int scale) {
```

Comment:
Do we really need this method? Maybe we can inline it?

Reply:
done